src/main/scala/io/pravega/connectors/spark/PravegaSourceProvider.scala

```diff
@@ -1,519 +1,571 @@
-            streamConfig.scalingPolicy(ScalingPolicy.byEventRate(targetRateEventsPerSec.toInt, scaleFactor.toInt, minSegments.toInt))
-          case _ =>
-            streamConfig.scalingPolicy(ScalingPolicy.fixed(minSegments.toInt))
-        }
-      case _ => streamConfig
-    }
-
-    // set retention policy
-    streamConfig = (retentionDurationMilliseconds, retentionSizeBytes) match {
-      case (Some(retentionDurationMilliseconds), None) =>
-        streamConfig.retentionPolicy(RetentionPolicy.byTime(Duration.ofMillis(retentionDurationMilliseconds.toLong)))
-      case (None, Some(retentionSizeBytes)) =>
-        streamConfig.retentionPolicy(RetentionPolicy.bySizeBytes(retentionSizeBytes.toLong))
-      case _ =>
-        streamConfig
-    }
-
-    log.info("streamConfig is {}", streamConfig)
-
-
-    return streamConfig.build()
-  }
-
-  def getPravegaStreamCut(
-                           params: Map[String, String],
-                           streamCutOptionKey: String,
-                           defaultStreamCut: PravegaStreamCut): PravegaStreamCut = {
-
-    params.get(streamCutOptionKey).map(_.trim) match {
-      case Some(offset) if offset.toLowerCase(Locale.ROOT) == STREAM_CUT_LATEST =>
-        LatestStreamCut
-      case Some(offset) if offset.toLowerCase(Locale.ROOT) == STREAM_CUT_EARLIEST =>
-        EarliestStreamCut
-      case Some(offset) if offset.toLowerCase(Locale.ROOT) == STREAM_CUT_UNBOUNDED =>
-        UnboundedStreamCut
-      case Some(base64String) => SpecificStreamCut(StreamCut.from(base64String))
-      case None => defaultStreamCut
-    }
-  }
-}
-
-object PravegaReader {
-  private[spark] val EVENT_FIELD_NAME = "event"
-  private[spark] val SCOPE_FIELD_NAME = "scope"
-  private[spark] val STREAM_FIELD_NAME = "stream"
-  private[spark] val SEGMENT_ID_FIELD_NAME = "segment_id"
-  private[spark] val OFFSET_FIELD_NAME = "offset"
-
-  private[spark] val pravegaSchema: StructType = StructType(Seq(
-    StructField(EVENT_FIELD_NAME, BinaryType),
-    StructField(SCOPE_FIELD_NAME, StringType),
-    StructField(STREAM_FIELD_NAME, StringType),
-    StructField(SEGMENT_ID_FIELD_NAME, LongType),
-    StructField(OFFSET_FIELD_NAME, LongType)
-  ))
-}
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.connectors.spark
+
+import java.net.URI
+import java.time.Duration
+import java.util
+import java.util.{Locale, Optional}
+
+import io.pravega.client.ClientConfig
+import io.pravega.client.admin.StreamManager
+import io.pravega.client.stream.{RetentionPolicy, ScalingPolicy, StreamConfiguration, StreamCut}
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.{AnalysisException, DataFrame, SQLContext, SaveMode}

```

Comment:
This import and others are unneeded. Use Intellij to automatically remove unneeded imports.